Update the Chart to the latest PSONO release

PeeK1e opened the pull request:
As proposed in #3 will do the Pull Request on the `dev` branch instead of `main`

Also fixed the issues that were mentioned in the discussions in #3

Regarding the extra secret-image, we need the `alpine/k8s` image to create the secrets on the cluster. To use a plain python image we would have to expand the python script to use k8s libraries. (That would be a task for another day)

---

## What changed
* We created a secret scraper, that allows to generate the psono secrets the first time you install PSONO
  * a helper container is used for that [psono-helm-secret-helper](https://github.com/cloudical-io/psono-helm-secret-helper)
 ```yaml
 # -- Secret Creation Config
# Set options to Generate a Secret on first install
# It will only run if you install the chart for the first time!
# The following Secrets will be created
# PSONO_SECRET_KEY
# PSONO_ACTIVATION_LINK_SECRET
# PSONO_DB_SECRET
# PSONO_EMAIL_SECRET_SALT
# PSONO_PRIVATE_KEY
# PSONO_PUBLIC_KEY
secrets:
  # Set to false if you have your own secrets.
  # If you have predefined values, you need to mount them in the sections below
  # or define them in the env section
  create: true
  # Set the Name of the generated secrets
  name: "psono-deploy-secrets"
  ```
* Secrets for the postgres can be referenced from a secret already in the cluster
```yaml
# -- Database credentials from Secret
databaseSecret:
  # -- enables or disables the secret to be used
  enabled: false
  # -- secret name
  name: "psono-postgres-secret"
  # -- Username Key Name
  usernameKey: "username"
  # -- Password Key Name
  passwordKey: "password"
  ```
  * The Ingress was updated to handle the new routing style of PSONOs' Server
    * It takes the `PSONO_URL_PREFIX` variable that was recently introduced into PSONO

PeeK1e committed:
Update: Ingress to work on Kubernetes >= 1.19.0

PeeK1e committed:
Add: pre-install helm hook to create PSONO secrets

PeeK1e committed:
Update: values.yaml to reflect secret generation

PeeK1e committed:
Update: some configs

PeeK1e committed:
Update: Server Deployment

PeeK1e committed:
Add: Database creds from secret

PeeK1e committed:
Update: Docs

PeeK1e committed:
Merge pull request #1 from cloudical-io/dev

Merge Dev into Main

PeeK1e committed:
fix: startup bug when using enterprise server

PeeK1e committed:
chore: bump chart version

PeeK1e committed:
add: liveness probe to server

PeeK1e committed:
remove: liveness probe from container, causes crashloop

PeeK1e committed:
update: changelog

PeeK1e committed:
fix: exposing secrets

PeeK1e commented:
i just saw i could just change branches in the edit drop down. I almost exclusively work on GitLab, sorry for the new PR ':D

Commit:
fix: urls in readme

PeeK1e committed:
Merge pull request #2 from cloudical-io/readme-fixes

fix: urls in readme

tiagoposse commented on templates/preflight-job.yaml:
```diff
@@ -0,0 +1,26 @@
+      automountServiceAccountToken: true
+      containers:
+        - name: secret-creator
+          image: harbor.cloudical.net/cci-tools/psono-helm-secret-helper:latest
```
Please pin this version, might break the chart in the future otherwise

PeeK1e replied:
finally had some time to fix this aswell :)

tiagoposse commented:
thanks for this!
small comment to prevent breaking in the future, ready to merge once done

Commit:
feat: ENV from secret
for webclient and adminClient incl. DB-secret

Commit:
feat: fixed envFrom issue
Independent if secret is rolled out from script or generated manually,
if named values will be used as env-variables for server.

Commit:
feat: Changelog added

PeeK1e committed:
update: changelog.md

PeeK1e committed:
Merge pull request #3 from cloudical-io/extend-envs

feat: ENV from secret

Commit:
feat!: added Ingress Classname to values.yaml

PeeK1e committed:
Merge pull request #4 from cloudical-io/add-ingressclass

feat!: added Ingress Classname to values.yaml

Commit:
feat: chartVersion

PeeK1e committed:
Merge pull request #5 from cloudical-io/correct-chartVersion

feat: chartVersion

PeeK1e committed:
update: ping image to commit version

PeeK1e commented:
There were some more bugfixes and improvements to the existing changes since we noticed some issues while rolling out the chart in a prod environment. But it should be done now :)

PeeK1e commented:
@tiagoposse how is it looking :)
